lib/dox.js

```diff
@@ -56,6 +56,84 @@ dox.contextPatternMatchers = [
     }// if constant
   },
 
+  //Exceptions
+  function (str) {
+    if (/(^[\w]+)(\s+)(exception)/i.exec(str)) {
+      // #40 fix for comments above a constant
+      str = str.replace(/^\s*--.*$/mg, '');
+
+      var
+        ret = {
+          type : 'exceptions',
+          header : str,
+          exceptions : []
+        },
+        exceptionsArr = str.split(';')
+        ;
+
+      exceptionsArr.forEach(function(exception){
+        // Comments after exceptions may still be in the array.
+        // As such only allow definitions of exceptions
+        if (/^\s*\w+\s+exception(?!_).*$/gi.test(exception)){
+          var
+            myException = {
+              name : '',
+              // Remove any "\n" and the begining
+              code : exception.replace(/^[\n]*/, '') + ';' //Need to re-append missing ";" since removed with .split
+            }
+            ;
+
+          /^\s*([\w]+)\s+exception(\s*)/i.exec(exception);
+          myException.name = RegExp.$1.trim();
+
+          ret.exceptions.push(myException);
+        }// if exception
+      });//exceptionsArr.forEach
+
+      return ret;
+    }// if exception
+  },
+
+  //Variables
+  function (str) {
+    if (str.substring(0, 4) != "type" && str.substring(0, 7) != "subtype" &&
+      /^\s*([\w]+)\s*(([\w]+)|([\w]+\(\s*([\d]+)(,\s*[\d]+)?\s*\)))((\s+)|;)((default|:=)(\s+)((\'[\w]*\')|[\d]+|[\w]+((\s+)[\w]+)?))?/i.exec(str)) {
```

Comment:
The reason why the regex is so large is because there are multiple ways to declare a variable. For instance we have the following ways to declare a variable:

l_number number;
l_number number(100);
l_number number(10, 2);
l_number number(10, 2) default 100.10;
l_string varchar2(100) default 'test variable';
l_object_type test_object := new test_object();

I'm not very good at writing regex so there is probably a much better way to find the variables inside the package headers. Currently im also not checking if type or subtype is not at the beginning (in the first part of the if statement).